exercises/04_linting/precipitation_climatology.py

```diff
@@ -0,0 +1,320 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import xarray as xr
+import scipy
+import cf_xarray
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+import cmocean
+import regionmask
+
+
+import matplotlib.ticker as mticker
+from cartopy.mpl.gridliner import LONGITUDE_FORMATTER, LATITUDE_FORMATTER
+
+
+def convert_precipitation_units(precipitation_in_kg_per_m_squared_s):
+    """Convert kg m-2 s-1 to mm day-1."""
+
+    precipitation_in_mm_per_day = xr.DataArray(
+        precipitation_in_kg_per_m_squared_s * 86400
+    )
```

Comment:
I'm not sure this is a clearer way to perform this operation when compared to referencing the `.data` attribute directly.
Would be keen to hear your reasons for taking this approach as I am not sure I am correct.
Perhaps @MarionBWeinzierl has some thoughts too?

Reply:
I made the decision to assign to a new variable which makes it so that the .data approach didn't work anymore. This was me trying to go for a more functional programming style and avoiding mutating an existing object, but I guess the object only exists within the scope of the function so it probably doesn't make too much difference. It also made the naming a bit easier, but naming it `precipitation_data_being_converted` or something like that could work too.

Reply:
Yep, all seems sensible.
Why do we need to use `xr.DataArray()`?

Reply:
I'm not too sure why I did that now, have removed it and all works fine :)